packages/kit/src/exports/vite/index.js

```diff
@@ -333,7 +333,25 @@ function kit({ svelte_config }) {
 	const plugin_virtual_modules = {
 		name: 'vite-plugin-sveltekit-virtual-modules',
 
-		async resolveId(id) {
+		async resolveId(id, importer) {
+			//If importing from a service-worker, only allow $service-worker & $env/static/public, but none of the other virtual modules.
+			//This check won't catch transitive imports, but it will warn when the import comes from a service-worker directly.
+			//Transitive imports will be caught during the build.
```

Comment:
```suggestion
			// If importing from a service-worker, only allow $service-worker & $env/static/public, but none of the other virtual modules.
			// This check won't catch transitive imports, but it will warn when the import comes from a service-worker directly.
			// Transitive imports will be caught during the build.
```